presto-main/src/main/java/com/facebook/presto/util/DateTimeUtils.java

```diff
@@ -117,14 +124,26 @@ public static String printDate(int days)
                 .withOffsetParsed();
     }
 
+    public static long parseTimestampLiteral(String value)
+    {
+        try {
+            DateTime dateTime = TIMESTAMP_WITH_TIME_ZONE_FORMATTER.parseDateTime(value);
+            return packDateTimeWithZone(dateTime);
+        }
+        catch (Exception e) {
```

Comment:
can we narrow the Exception type hierarchy space here and in other places? IMO the best would be to avoid using try-catch to control the flow.

Reply:
I agree - though concept was not introduced with this PR. leaving for later.